internal/controller/controllers/agent_controller.go

```diff
@@ -164,6 +173,10 @@ func (r *AgentReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (
 		log.WithError(err).Warnf("failed to set infraEnv name label on agent %s/%s", agent.Namespace, agent.Name)
 	}
 
+	if err = r.setStateAnnotation(ctx, log, h, agent); err != nil {
```

Comment:
Should we maybe do this in the same place we update the status?

Reply:
Sorry on my first pass I thought this was an entire copy of the status (like BMH) and then it made more sense to save it to an annotation at the same time we updated it in the CR.

Reply:
np, I'll revert to the previous approach:)